dozer/cogs/_utils.py

```diff
@@ -289,4 +301,56 @@ def decorator(func):
             func.__required_permissions__ = discord.Permissions()
             func.__required_permissions__.update(**required)
         return func
+
     return decorator
+
+
+class PrefixHandler:
+    """Handles dynamic prefixes"""
+
+    def __init__(self, default_prefix):
+        self.default_prefix = default_prefix
+        self.prefix_cache = {}
+
+    def handler(self, bot, message):
+        """Process the dynamic prefix for each message"""
+        dynamic = self.prefix_cache.get(message.guild.id)
```

Comment:
When a command is run in DMs, `message.guild` is `None`. This needs to handle that case.